aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -909,8 +909,8 @@ Tensor matmul_nested(const Tensor& self, const Tensor& mat2) {
     if (batch_size == 1) {
       at::mm_out(
           output_unbind[i],
-          self_buffer.as_strided(self_size, self_strides[i], self_offsets[i]),
-          mat2_buffer.as_strided(mat2_size, mat2_strides[i], mat2_offsets[i])
+          self_buffer.as_strided(self_size, self_strides[i], self_offsets[i]).squeeze(0),
```

Comment:
Wait, doesn't this *require* the input to be 4-dim? What happens if you pass a 5-dim NT?

In general, let's replace these for loops by a call to `to_padded`, `matmul`, `from_padded` until we have a more performant kernel. All these kernel launches are very inefficient.

Reply:
Oh you are right.